### PR TITLE
[10.0][MIG] stock_warehouse_orderpoint_stock_info_unreserved

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Standard migration.

### CODE REVIEW

**Review by JordiBForgeFlow — APPROVED** [review]

**Review by LoisRForgeFlow — APPROVED** [review]
Tested, LGTM :+1:

**Review by Cedric-Pigeon — APPROVED** [review]
Code review

**Review by serpentcs-dev1 — APPROVED** [review]
Code & Funcional Review

LGTM :+1:

File: stock_warehouse_orderpoint_stock_info_unreserved/models/stock_warehouse_orderpoint.py
Comment (fmdl) [inline stock_warehouse_orderpoint_stock_info_unreserved/models/stock_warehouse_orderpoint.py:14]: Dear I have made a PR to compute fast https://github.com/OCA/stock-logistics-warehouse/pull/340/

Please find a code to replace, like that the methode `_product_available()` (replaced by `_compute_quantities_dict()` in new version), is computed once by location.

```
   def _compute_product_available_qty(self):
        super(StockWarehouseOrderpoint, self)._compute_product_available_qty()
        op_by_loc = defaultdict(lambda: self.env['stock.warehouse.orderpoint'])
        for order in self:
            op_by_loc[order.location_id] |= order
        for location_id, order_in_loc in op_by_loc.items():
            products = order_in_loc.mapped('product_id').with_context(
                location=location_id.id)._compute_quantities_dict(
                lot_id=None, owner_id=None, package_id=None)
            for order in order_in_loc:
                product = products[order.product_id.id]
                order.product_location_qty_available_not_res = products['qty_available_not_res']
```

cc @sylvain-garancher

**Comment by fmdl** [comment]
Idiot question : why this module is not merge with https://github.com/fmdl/stock-logistics-warehouse/tree/10.0/stock_warehouse_orderpoint_stock_info

**Review by rousseldenis — APPROVED** [review]
Code review

**Comment by JordiBForgeFlow** [comment]
@hveficent can you review?

**Review by HviorForgeFlow — COMMENTED** [review]
I've got an error when I was trying to create a product. This is the error I got.

image

File "/home/odoo/build/OCA/stock-logistics-warehouse/stock_available/models/product_template.py", line 70, in _compute_potential_qty
    tmpl.potential_qty = res[tmpl.id]['potential_qty']
KeyError: 'potential_qty'

**Comment by rousseldenis** [comment]
@hveficent @mreficent @jbeficent The stock_available computation has been refactored in https://github.com/OCA/stock-logistics-warehouse/pull/305/

Maybe we can move forward on that PR ?

File: stock_warehouse_orderpoint_stock_info_unreserved/models/stock_warehouse_orderpoint.py
Comment (HviorForgeFlow) [inline stock_warehouse_orderpoint_stock_info_unreserved/models/stock_warehouse_orderpoint.py]: Fields must be declared on the beginning of the class before methods. https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#models

Comment (MiquelRForgeFlow) [reply stock_warehouse_orderpoint_stock_info_unreserved/models/stock_warehouse_orderpoint.py]: Done

File: stock_warehouse_orderpoint_stock_info_unreserved/models/stock_warehouse_orderpoint.py
Comment (MiquelRForgeFlow) [reply stock_warehouse_orderpoint_stock_info_unreserved/models/stock_warehouse_orderpoint.py:14]: Done

**Comment by MiquelRForgeFlow** [comment]
@rousseldenis , we are thinking in merging this PR because it has the needed approvals and we need to advance forward. The PR https://github.com/OCA/stock-logistics-warehouse/pull/305 you mentioned before seems old and abandoned and doesn't have the approvals. Thus, is it ok for you if we proceed to merge this PR now?

**Comment by rousseldenis** [comment]
@mreficent Ok for me. I will take care of #305 to make it evolve after.

**Comment by LoisRForgeFlow** [comment]
@mreficent @rousseldenis Thanks!

**Comment by MiquelRForgeFlow** [comment]
@webtekindo Can you give more details? I tested now in runbot opening the create form of product and gives the following error:

[stock_available/models/product_template.py](https://github.com/OCA/stock-logistics-warehouse/blob/10.0/stock_available/models/product_template.py), line 70
KeyError: 'potential_qty'
tmpl.potential_qty = res[tmpl.id]['potential_qty']

**Pull request merged**
